### PR TITLE
keep disable_fp16_compression rt_info

### PR DESCRIPTION
**Pull request by pavel-esir**
### Details:
In these notebooks to prevent accuracy drop https://github.com/openvinotoolkit/openvino_notebooks/pull/1382 we use fp16 calibration utility in order to mark and keep keep some sensitive nodes in the original fp32 precision. Since calibration takes ~2-3 mins to complete on each model we need to serialize this marking information into IR.

We use `'precise'` rt_info for that, in accordance with agreements on the API 2.0 architecture meeting (IE_API.pptx presentation slide 17).

How it looks in the IR after serialization
image

### CODE REVIEW

**Comment by slyalin** [comment]
Why do we need it?

**Comment by pavel-esir** [comment]
> Why do we need it?

Without these changes, rt_info which i set during calibrating/upcasting to fp32 is not saved into IR

File: src/core/src/graph_util.cpp [hunk 1]
```diff
@@ -335,6 +336,12 @@ void serialize(const std::shared_ptr<const ov::Model>& m,
                const std::string& bin_path,
                ov::pass::Serialize::Version version) {
     ov::pass::Manager manager;
+    // TODO: if rt_info is set in python api as a string ['disable_fp16_compression_0'] = '',
```
Comment (itikhono) [inline src/core/src/graph_util.cpp:339]: @jiwaszki do we have plans to export RuntimeAttribute and inherited classes to python API?
it looks like this is similar to CVS-105807

Comment (jiwaszki) [reply src/core/src/graph_util.cpp]: There are no advances in Core itself. As for now Python API is blocked.

Comment (itikhono) [reply src/core/src/graph_util.cpp]: @jiwaszki do you agree with the current workaround?

Comment (jiwaszki) [reply src/core/src/graph_util.cpp]: I don't have anything against it.

CC @akuporos what do you think?

**Review by jiwaszki — APPROVED** [review]

**Review by slyalin — CHANGES_REQUESTED** [review]
Explain why it is required. This flag wasn't designed to be serialized.

**Comment by pavel-esir** [comment]
> Explain why it is required. This flag wasn't designed to be serialized.

Besides using it in ov transformations I reused this rt_info in fp16 calibration python utility here https://github.com/openvinotoolkit/openvino_notebooks/pull/1382. Calibration takes ~2-3 mins to complete on each model. If we don't serialize this rt_info into IR then we would need to run calibration every time we rerun notebook.

Yes, originally it wasn't designed to be serialized, but i don't see any critical problems with serializing it. The only collateral 'damage' of this changes, it that affects `openvino.convert_model` in a way that in fp16_compressed IRs now have this rt_info on shape of nodes, but this does not cause any troubles at all when they are read, this marking does not leak any where and does not lead to adding extra converts.

Do you think it's critical and we should consider other solutions? Can you give example where this could lead to problems?

**Review by akuporos — APPROVED** [review]

File: src/core/src/pass/serialize.cpp [hunk 2]
```diff
@@ -1194,6 +1194,14 @@ void serializeFunc(std::ostream& xml_file,
 namespace ov {
 bool pass::Serialize::run_on_model(const std::shared_ptr<ov::Model>& model) {
     RUN_ON_FUNCTION_SCOPE(Serialize);
+
+    // TODO: if rt_info is set in python api as a string ['disable_fp16_compression_0'] = '',
```
Comment (itikhono) [inline src/core/src/pass/serialize.cpp:1198]: I would recommend to add a jira ticket (new or already existing)

**Review by itikhono — APPROVED** [review]

File: src/common/transformations/include/transformations/rt_info/disable_fp16_compression.hpp [hunk 3]
```diff
@@ -25,14 +25,18 @@ TRANSFORMATIONS_API void do_not_postpone_fp16_compression(RTMap& rt_info);
 /**
  * @ingroup ie_runtime_attr_api
  * @brief DisableFP16Compression class represents runtime info attribute that marks operation
- * as prohibitted to convert to FP16 as part of Compressed Only format.
+ * as prohibited to convert to FP16 precision and they should be inferred in the precisely original precision.
```
Comment (slyalin) [inline src/common/transformations/include/transformations/rt_info/disable_fp16_compression.hpp:28]: When you rename the class (not in this PR) please update the comment to extend it beyond FP16 mentioning FP16 as an example.

File: src/core/src/pass/serialize.cpp [hunk 4]
```diff
@@ -1208,6 +1208,14 @@ void serializeFunc(std::ostream& xml_file,
 namespace ov {
 bool pass::Serialize::run_on_model(const std::shared_ptr<ov::Model>& model) {
     RUN_ON_FUNCTION_SCOPE(Serialize);
+
+    // TODO xxx-105807: if rt_info is set in python api as a string ['disable_fp16_compression_0'] = '',
```
Comment (slyalin) [inline src/core/src/pass/serialize.cpp:1212]: Name of attribute is different.

Comment (slyalin) [reply src/core/src/pass/serialize.cpp:1212]: `_0` looks weird, can we avoid it?

**Review by slyalin — CHANGES_REQUESTED** [review]

File: src/core/src/pass/serialize.cpp (hunk 4 above)
Comment (pavel-esir) [reply src/core/src/pass/serialize.cpp:1212]: I will update the name. But unfortunately we can not get rid of `_0`

File: src/common/transformations/include/transformations/rt_info/disable_fp16_compression.hpp (hunk 3 above)
Comment (pavel-esir) [reply src/common/transformations/include/transformations/rt_info/disable_fp16_compression.hpp]: updated in the current PR: `represents runtime info attribute that marks operation as prohibited to convert to lower precision (e.g. to FP16) and they should be inferred precisely in the original precision`.

File: src/core/src/pass/serialize.cpp (hunk 4 above)
Comment (pavel-esir) [reply src/core/src/pass/serialize.cpp]: updated the name

File: src/common/transformations/include/transformations/rt_info/disable_fp16_compression.hpp [hunk 5]
```diff
@@ -25,14 +25,18 @@ TRANSFORMATIONS_API void do_not_postpone_fp16_compression(RTMap& rt_info);
...
  */
 class TRANSFORMATIONS_API DisableFP16Compression : public RuntimeAttribute {
 public:
-    OPENVINO_RTTI("disable_fp16_compression", "0");
+    OPENVINO_RTTI("precise", "0");
 
     DisableFP16Compression() = default;
 
+    bool visit_attributes(AttributeVisitor& visitor) override {
+        return true;
```
Comment (slyalin) [inline src/common/transformations/include/transformations/rt_info/disable_fp16_compression.hpp:38]: Just for my education: does such implementation lead to missing `value` filed in IR as in the example in the description? Is it a "standard" way to implement Boolean attributes like this one?

Comment (pavel-esir) [reply src/common/transformations/include/transformations/rt_info/disable_fp16_compression.hpp:38]: As far as i understood yes. E.g. existing `'decompression'` rt_info, i've seen the same thing
```
<layer id="4" name="bert/embeddings/word_embeddings" type="Convert" version="opset1">
    <data destination_type="f32" />
    <rt_info>
         <attribute name="decompression" version="0" />
    </rt_info>
```

**Review by slyalin — APPROVED** [review]

**Pull request merged**